src/functions/Profile/ProfilePage.js

```diff
@@ -0,0 +1,136 @@
+        console.log("printing user data: ",this.userData);
+        return this.userData;
+    }
+    getBio(){
+        console.log("printing bio: ",this.bio);
+        return this.bio;
+    }
+
+    getGenres(){
+        console.log("printing genres: ",this.genres);
+        return this.genres;
+    }
+
+    getPronouns(){
+        console.log("printing pronouns: ",this.pronouns);
+        return this.pronouns;
+    }
+
+    getUsername(){
+        console.log("printing username: ",this.username);
+        return this.username;
+    }
+
+    async initialize() {
+        await this.setProfilePicture();
+    }
+
+    async setProfilePicture() {
+        try {
+            const fetchedProfilePicture = await fetchPFP(this.uid);
+            if (fetchedProfilePicture) {
+                this.profile_picture = fetchedProfilePicture;
+                console.log("Profile picture fetched successfully:", this.profile_picture);
+            } else {
+                console.error("Failed to fetch profile picture");
+            }
+        } catch (error) {
+            console.error("Error fetching profile picture:", error);
+        }
+    }
+
+    getProfilePicture() {
+        console.log("Printing profile picture:", this.profile_picture);
+        return this.profile_picture;
+    }
+
+    getFollowersCount(){
+        console.log("Printing Followers count: ", this.followersCount);
+        return this.followersCount;
+    }
+
+    getFollowingCount(){
+        console.log("Printing Followers count: ", this.followingCount);
+        return this.followingCount;
+    }
+
+    getFollowers(){
+        console.log("Printing Followers: ", this.followers);
+        return this.followers;
+    }
+
+    getFollowing(){
+        console.log("Printing Followers: ", this.following);
+        return this.following;
+    }
+    getIsFollowing(){
+        console.log("Printing isFollowing: ", this.isFollowing);
+        return this.isFollowing;
+    }
+    setUserData = async() =>{
+        try{
+        const userRef = doc(db, 'users', this.uid);
+        const docSnapshot = await getDoc(userRef);
+        if (docSnapshot.exists()) {
+            console.log("started");
+            this.userData = docSnapshot.data();
+            this.bio = docSnapshot.data().bio;
+            this.genres = docSnapshot.data().genres;
+            this.pronouns = docSnapshot.data().pronouns;
+            this.username = docSnapshot.data().username;
+            console.log("username from class :",this.username);
+            this.followers = docSnapshot.data().followers;
+            this.following = docSnapshot.data().following;
+            this.followersCount = this.userData.followers?.length || 0;
+            this.followingCount = this.userData.following?.length || 0;
+            this.isFollowing = this.userData.following?.includes(this.uid);
+            //    userData.followers?.includes(currentUser.uid)
+            console.log("user data fetched successfully");
```

Comment:
These logs seem like they are performing the same action as the try catch function that you have wrapping this get func

Reply:
I removed the else. Try-catch is used to catch any exceptions in this case.